src/pf-modal/pf-modal.component.js

```diff
@@ -0,0 +1,250 @@
+    this._mask = null;
+  }
+
+  /*
+   * Called when an instance was inserted into the document
+   * So we place event bindings here
+   */
+  attachedCallback () {
+    if (this.open) {
+      this._showModal();
+    }
+    this._target.addEventListener('click', this.show.bind(this), false);
+    this._element.addEventListener('click', this.hide, false);
+    if (this.keyboard) {
+      this._element.addEventListener('keydown', this._keydownHandler, false);
+    }
+  }
+
+  /*
+   * Show the modal when attribute open is added or property open is enabled
+   * @private
+   */
+  _showModal () {
+    pfUtil.addClass(document.body, 'modal-open');
+    this._mask = document.createElement('div');
+    this._mask.className = 'modal-backdrop fade';
+    document.body.appendChild(this._mask);
+    pfUtil.reflow(this._mask);
+    pfUtil.addClass(this._mask, 'in');
+    pfUtil.reflow(this._element);
+    pfUtil.addClass(this._element, 'in');
+    pfUtil.once(this._dialog, 'transitionend', this._afterShowModal, this);
+  }
+
+  /*
+   * Show the modal
+   * @public
+   */
+  show () {
+    this._element.dispatchEvent(new CustomEvent('show.bs.modal', { 'detail': { 'relatedTarget': this._target }}));
+    if (this.open) {
+      return;
+    }
+    this.open = true;
+  }
+
+  /*
+   * Callback after modal is shown
+   * @private
+   */
+  _afterShowModal () {
+    this._element.focus();
+    this._element.dispatchEvent(new CustomEvent('shown.bs.modal', { 'detail': { 'relatedTarget': this._target }}));
+  }
+
+  /*
+   * Hide the modal when attribute open is removed or property open is disabled
+   * @private
+   */
+  _hideModal () {
+    pfUtil.removeClass(document.body, 'modal-open');
+    pfUtil.removeClass(this._mask, 'in');
+    pfUtil.removeClass(this._element, 'in');
+    pfUtil.once(this._element, 'transitionend', this._afterHideModal, this);
+  }
+
+  /*
+   * Hide the modal
+   * @public
+   */
+  hide (e) {
+    if (e) {
+      e.preventDefault();
+      if (!(pfUtil.getClosest(e.target, '.pf-hide-modal') || (e.target === e.currentTarget && this.backdrop))) {
+        return;
+      }
+    }
+
+    this._element.dispatchEvent(new CustomEvent('hide.bs.modal', {}));
+
+    if (!this.open) {
+      return;
+    }
+    this._hideModal();
+  }
+
+  /*
+   * Callback after the modal is hidden
+   * @private
+   */
+  _afterHideModal () {
+    this.removeAttribute('open');
+    if (this._mask) {
+      this._mask.remove();
+      this._mask = null;
+    }
+    this._element.dispatchEvent(new CustomEvent('hidden.bs.modal', {}));
+  }
+
+  /*
+   * Toggle the visible/invisible state of modal
+   * @public
+   */
+  toggle () {
+    return this.open ? this.hide() : this.show();
+  }
+
+  /*
+   * Handler of keydown event of escape key
+   * @private
+   */
+  _keydownHandler (e) {
+    let isEscape = (e.key && e.key === 'Escape') || (e.keyIdentifier && e.keyIdentifier === 'U+001B') || (e.keyCode && e.keyCode === 27);
```

Comment:
Many Angular developers won't use the components because of jQuery -- it's a non-starter for some people.